eth-contracts/scripts/fuzzyTest/1_EaglesongFuzzy.test.js

```diff
@@ -0,0 +1,44 @@
+const { expect } = require('chai')
+const vectors = require('../../test/data/testEaglesongFuzzy.json')
+const log = console.log.bind(console)
+const { calculateEaglesong } = vectors
+
+contract('EaglesongFuzzy', () => {
+  let eaglesong, testEaglesong
+  let provider
+
+  before(async function () {
+    // disable timeout
+    this.timeout(0)
+
+    // deploy hamstersong
+    let factory = await ethers.getContractFactory(
+      'contracts/Eaglesong.sol:Eaglesong'
+    )
+    eaglesong = await factory.deploy()
+    await eaglesong.deployTransaction.wait(1)
+    const hamAddr = eaglesong.address
+
+    // deploy TestEaglesong
+    factory = await ethers.getContractFactory(
+      'contracts/test/TestEaglesong.sol:TestEaglesong'
+    )
+    testEaglesong = await factory.deploy(hamAddr)
+    await testEaglesong.deployTransaction.wait(1)
+    provider = testEaglesong.provider
+  })
+
+  describe('Eaglesong correct case', async function () {
```

Comment:
The given data set is not random and big enough.
Should use random to generate data to hash and compare with other version (e.g. Rust version).

Reply:
It is finished before on my other branch.  Test data is generated randomly by Rust code.  
More test data will cause hardhat test slowly, so I just put this little tests on the pr. 
Actually eaglesong random tests have been finished more than 100,000 times locally

- [test generator by rust](
https://github.com/cosinlink/ckb-spv-rust/blob/a3b1990521920328de8b7eccd4b03875d6a5492c/src/generate_sol_test.rs#L227)